File: Python/69. Sqrt(x).py
class Solution:
    def mySqrt(self, x: int) -> int:
        if x <= 1: return x
        high, low = x // 2, 0
        while high >= low:
            mid = (high + low) // 2
            temp = mid * mid
            if temp == x:
                return mid
            elif temp > x:
                high = mid - 1
            else:
                low = mid + 1
        return high

File: C/69. Sqrt(x).c
int mySqrt(int x) {
    if (x == 1) return 1;

    long long i = 0, j = x / 2;

    while (i <= j) {
        long long m = (i + j) / 2;
        long long t = m * m;

        if (t == x) return m;
        else if (t > x) j = m - 1;
        else i = m + 1;
    }

    return j;
}
